src/Http/Headers/src/CookieHeaderParser.cs

```diff
@@ -13,7 +14,7 @@ internal CookieHeaderParser(bool supportsMultipleValues)
         {
         }
 
-        public override bool TryParseValue(StringSegment value, ref int index, out CookieHeaderValue parsedValue)
+        public override bool TryParseValue(StringSegment value, ref int index, [MaybeNull] out CookieHeaderValue? parsedValue)
```

Comment:
Do we need both `[MaybeNull]` and `?`? Should this also use `[NotNullWhen(true)]`?

Reply:
Removing `MaybeNull` would be the way to go. This can remove a null value while returning true, so `NotNullWhen` doesn't quite work.